openhcl/hcl/src/ioctl.rs

```diff
@@ -1244,7 +1242,30 @@ impl MshvHvcall {
                 todo!();
             }
             Some(Vtl::Vtl0) => {
-                assert!(matches!(name, HvX64RegisterName::GuestOsId));
+                // Only VTL-private registers can go through this path.
+                // VTL-shared registers have to go through the kernel (either
+                // via the CPU context page or via the dedicated ioctl), as
+                // they may require special handling there.
+                //
+                // Register access should go through the register page if
+                // possible (as a performance optimization), so the set of
+                // registers handled here should be small. Except for
+                // GuestOsId and Sint2, in practice these registers are handled
+                // here only if the register page is unavailable (e.g., running
+                // in WHP).
+                assert!(matches!(
+                    name,
+                    HvX64RegisterName::GuestOsId
+                        | HvX64RegisterName::Cr0
+                        | HvX64RegisterName::Efer
+                        | HvX64RegisterName::Rsp
+                        | HvX64RegisterName::Ds
+                        | HvX64RegisterName::Es
+                        | HvX64RegisterName::Fs
+                        | HvX64RegisterName::Gs
+                        | HvX64RegisterName::Ss
+                        | HvX64RegisterName::Sint2
+                ));
```

Comment:
Open question: should this assert capture every register that might come through here (e.g., the rest of the Sint registers), or just the ones that we observe in practice?

Reply:
Probably should have the ones that we'd reasonably expect, so that we're not likely to crash on say a new windows version that touches a new register or something

Reply:
That sounds good, though I don't know how to compute "the ones that we'd reasonably expect". For WHP, perhaps it could be anything that normally lives on the register page, which gets to be a large-ish set of registers. I'll start with that though and see how it looks.

Reply:
We don't have a path where the guest can ask for an arbitrary register and then we go down to this path to retrieve it, do we? Or is that WHP? I was assuming that we would be able to just include the ones we have effectively hardcoded, and then we can just update this list as we add more.